### PR TITLE
also use color ramp shader widget for pseudo color renderer

### PR DESCRIPTION
**Pull request by PeterPetrik**
Use QgsColorRampShaderWidget  for raster layers (it is already used for mesh layers)

### CODE REVIEW

**Comment by nirvn** [comment]
Please give me 24 hours to review this. Thanks

File: src/gui/raster/qgscolorrampshaderwidget.h [hunk 1]
```diff
@@ -51,15 +51,27 @@ class GUI_EXPORT QgsColorRampShaderWidget: public QWidget, protected Ui::QgsColo
...
+    void setExtent( const QgsRectangle &extent );
 
     //! Sets min max and classify color tree
     void setMinMaxAndClassify( double min, double max );
 
     //! Sets min max
     void setMinMax( double min, double max );
 
+    //! Gets min value
+    double min() const;
```
[the remark this answers is not in the document]
Comment (PeterPetrik) [reply src/gui/raster/qgscolorrampshaderwidget.h]: np, thanks. Does this apply to all functions or just getters, let say `setMinMaxAndClassify` -> `setMinimumMaximumAndClassify` ?

[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/gui/raster/qgscolorrampshaderwidget.h]: Yes, if that's not already part of stable API (we didn't get them all for 3.0)

**Comment by nirvn** [comment]
OK, no request for change, good work.

**Comment by nirvn** [comment]
@PeterPetrik , beyond the code, which looked fine, I just compiled a build including these changes and couldn't spot any regressions. Congrats.

As I mentioned in another PR, if you want to go the extra mile here (and I'm crossing fingers you are :smile:), it'd be fantastic to add a list-to-ramp function, whereas changes to colors and steps in the list would update the color ramp widget. IMHO, that'd be a _huge_ UX improvement (it's been reported a couple of times on our issue tracker already).

**Comment by PeterPetrik** [comment]
@nirvn thanks a lot for review. Unfortunately I will be out most part of July, so I will not have time for it immediately. 

Would it be possible to merge this one (I do not have write/merge access) for now?

**Comment by nyalldawson** [comment]
@PeterPetrik 

This PR has introduced a serious regression. Steps to reproduce:

1. load a raster
2. style with singleband pseudo color renderer
3. in the layer styling dock, expand min/max settings and try to change the min/max mode from user defined to anything else. 

The mode is instantly reset to user, and there's no way to change any of the other min/max settings

Here's a cast:

peek 2018-07-10 14-09

**Comment by nirvn** [comment]
@PeterPetrik , regressions spotted:

Regression A:
- Add a raster to a new project, open the style dock
- Switch the renderer to singleband pseudocolor
- Expand the min max values settings groupbox, and select "(*) User defined"
- Modified the max value via line edit
- You'll notice while the list is updated, the actually renderer isn't, canvas doesn't change

Regression B:
- Add a raster to a new project, open the style dock
- Switch the renderer to singleband pseudocolor
- Manually modified the max value via its line edit (that'll automatically switch min/max settings to user defined)
- You'll notice only the first keystroke in the line edit is actually reflected by the renderer. I.e., if I have a DEM and I type enter 300 as max value (i.e. key strokes 3, 0, and 0), the renderer itself will only update for the keystroke "3", not the additional "0" (which would become 30) and "0" (which would become 300)

**Comment by PeterPetrik** [comment]
ups, let me try to fix it

**Comment by nyalldawson** [comment]
@PeterPetrik I've found a regression here -- it's no longer possible to set the suffix text for legend items in a psuedo color renderer as a result of this work, now it's just ignored and not used anywhere. I couldn't follow the logic in QgsColorRampShaderWidget::autoLabel() -- what purpose is the grey color used for here?

**Comment by PeterPetrik** [comment]
@nyalldawson The code is mostly copied from previous implementation. Gray colored text is autogenated, black one should be manually edited by user (so we honour the user decision).

**Comment by nyalldawson** [comment]
@peterpetrik I can't see how the text would ever become gray to begin with -- the only way i can get that to happen is by manually deleting all the existing legend text. Nothing else I do causes the text to become grey -- even totally resetting the renderer by deleting all classes and then re-classifying results in black text with no suffix applied.

**Pull request merged**